### PR TITLE
Add policy to checklist

### PR DESCRIPTION
**Pull request by chartgerink**
This PR adds three items to the checklist based on some of the updated policies (introduced because of PMC requirements).

I previously discussed this change with @arfon (see https://github.com/openjournals/joss/pull/1106#issuecomment-1214316823) - I am happy to further discuss here (e.g., where in the checklist these should go 😊).

### CODE REVIEW

File: .buffy/templates/reviewer_checklist.md [hunk 1]
```diff
@@ -14,6 +14,10 @@
 - [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
 - [ ] **Contribution and authorship:** Has the submitting author ({{author-handle}}) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
 - [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)
+- [ ] **Data sharing:** If the paper contains original data, data are accessible to the reviewers.
+- [ ] **Data sharing:** If the paper contains original results, results are entirely reproducible by reviewers
```
Comment (danielskatz) [inline .buffy/templates/reviewer_checklist.md:18]: The heading here is wrong - maybe this should be reproducibility?  Also, I don't think "entirely" is always possible.  Finally, perhaps add "If the paper contains no original results, please check this item"

File: .buffy/templates/reviewer_checklist.md [hunk 2]
```diff
@@ -14,6 +14,10 @@
 - [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
 - [ ] **Contribution and authorship:** Has the submitting author ({{author-handle}}) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
 - [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)
+- [ ] **Data sharing:** If the paper contains original data, data are accessible to the reviewers.
```
Comment (danielskatz) [inline .buffy/templates/reviewer_checklist.md:17]: Perhaps add "If the paper contains no original data, please check this item"

File: .buffy/templates/reviewer_checklist.md (hunk 1 above)
Comment (chartgerink) [reply .buffy/templates/reviewer_checklist.md:18]: Thanks @danielskatz! Fixing now.

File: .buffy/templates/reviewer_checklist.md (hunk 2 above)
Comment (chartgerink) [reply .buffy/templates/reviewer_checklist.md:17]: Thanks! That's perfect.

**Review by arfon — APPROVED** [review]

**Pull request merged**